packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -32,14 +46,144 @@ export interface IEcsDeploymentGroup extends cdk.IResource {
 }
 
 /**
- * Note: This class currently stands as a namespaced container for importing an ECS
- * Deployment Group defined outside the CDK app until CloudFormation supports provisioning
- * ECS Deployment Groups. Until then it is closed (private constructor) and does not
- * extend {@link Construct}.
+ * Details about an ECS service and its corresponding ECS cluster
+ */
+export type EcsService = ecs.Ec2Service | ecs.FargateService | ecs.ExternalService;
+
+/**
+ * A route with an ELB listener and target group.
+ */
+export interface EcsTrafficRoute {
+  /**
+   * The load balancer for the traffic route.
+   *
+   * @default - blank if empty on test traffic route. Required for prod traffic route.
+   */
+  readonly listener: elbv2.IApplicationListener | elbv2.INetworkListener | undefined;
```

Comment:
I ended up creating a new IListener interface to avoid using a union type here:
https://github.com/clareliguori/aws-cdk/commit/c5eb191393cbc017afe0da2b758e56a1fb8bef9a#diff-1ce9801f6492a5d0a765325bd3737f3622605d4c7a6910547e075ce01ca06fcb